pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -103,7 +103,7 @@ func (s *Scheduler) getSchedules(ctx context.Context, constraints *v1alpha5.Cons
 			GPURequests v1.ResourceList
 		}{
 			Constraints: tightened,
-			GPURequests: resources.GPULimitsFor(pod),
+			GPURequests: resources.NonGenericResources(pod),
```

Comment:
so this could be more than just `GPURequests` right?  I guess the thought is that it doesn't matter with the current scheduler implementation?  Can we change the name of this variable?

Reply:
Fixed the name, I do like providing a comparison function better.  I suggested to Ellis below about a compare that provides both ordering & incompatibility information.